src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -880,10 +882,12 @@ public void visitIdent(JCIdent tree) {
                         result = thisValue();
                     } else {
                         FieldDesc fd = symbolToFieldDesc(sym, symbolSiteType(sym));
+                        // @@@ for now, for field of generic type, result type is the upper bound
```

Comment:
When we have an unqualified access to a field of generic type, I was expecting the type variable to be the result type of the field.load operation (or something that indicates the field is of generic type), but currently the result type is the upper bound of the type variable. Maybe in the new way modeling of types, we can have that.

Reply:
Ah, yes we don't model type variables (yet?). You can remove that comment if you wish, since that mapping of variable to upper bound is generally applied.